crates/pop-cli/src/commands/install/mod.rs

```diff
@@ -2,32 +2,234 @@
 
 use anyhow::Context;
 use clap::Args;
+use cliclack::{confirm, log, outro};
+use duct::cmd;
+use os_info::Type;
 use tokio::{fs, process::Command};
 
 #[derive(Args)]
 #[command(args_conflicts_with_subcommands = true)]
-/// Setup user environment for substrate development
-/// Runs script `scripts/get_substrate.sh`
-pub(crate) struct InstallArgs;
+/// Setup user environment for development
+pub(crate) struct InstallArgs {
+	/// Before install all the dependencies needed, do not ask the user for confirmation.
+	#[clap(short('y'), long)]
+	skip_confirm: bool,
+}
 
 impl InstallArgs {
 	pub(crate) async fn execute(self) -> anyhow::Result<()> {
-		if cfg!(target_os = "windows") {
-			return Ok(cliclack::log::error("Windows is supported only with WSL2")?);
+		if cfg!(target_os = "macos") {
+			log::info("ℹ️ Mac OS (Darwin) detected.")?;
+			install_mac(self.skip_confirm).await?;
+		} else if cfg!(target_os = "linux") {
+			match os_info::get().os_type() {
+				Type::Arch => {
+					log::info("ℹ️ Arch Linux detected.")?;
+					install_arch(self.skip_confirm).await?;
+				},
+				Type::Debian => {
+					log::info("ℹ️ Debian Linux detected.")?;
+					install_debian(self.skip_confirm).await?;
+				},
+				Type::Redhat => {
+					log::info("ℹ️ Redhat Linux detected.")?;
+					install_redhat(self.skip_confirm).await?;
+				},
+				Type::Ubuntu => {
+					log::info("ℹ️ Ubuntu detected.")?;
+					install_ubuntu(self.skip_confirm).await?;
+				},
+				_ => return not_supported_message(),
+			}
+		} else {
+			return not_supported_message();
 		}
-		let temp = tempfile::tempdir()?;
-		let scripts_path = temp.path().join("get_substrate.sh");
-		let client = reqwest::Client::new();
-		let script = client
-			.get("https://raw.githubusercontent.com/r0gue-io/pop-cli/pop-install/scripts/get_substrate.sh")
-			.send()
-			.await
-			.context("Network Error: Failed to fetch script from Github")?
-			.text()
-			.await?;
-		fs::write(scripts_path.as_path(), script).await?;
-		Command::new("bash").arg(scripts_path).status().await?;
-		temp.close()?;
+		install_rustup().await?;
+		log::success("✅ Installation complete.")?;
 		Ok(())
 	}
 }
+
+async fn install_mac(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/macos/")?;
+	if !skip_confirm {
+		prompt_for_confirmation("Homebrew, protobuf, openssl, rustup, and cmake")?
```

Comment:
Fine for now, but these static strings are error prone when updating n the future, easy to forget something when all scattered about and then things get out of sync and quality degrades.

Reply:
Done, good to follow good practices since the beginning!